src/websocket/base.ts

```diff
@@ -146,6 +145,29 @@ abstract class BaseWebSocket extends EventEmitter {
     this.ws.onmessage = (evt) => this.onMessage(evt);
     this.ws.onerror = (evt) => this.onError(evt);
     this.ws.onclose = () => this.onClose();
+
+    this.started = true;
+  }
+
+  /**
+   * Sets all event listeners to noops on the WebSocket instance 
+   * and close it.
+   **/
+  closeWs() {
+    if (!this.ws) {
+      return;
+    }
+
+    this.ws.onopen = () => {};
+    this.ws.onclose = () => {};
+    this.ws.onerror = () => {};
+    this.ws.onmessage = () => {};
+
+    if (this.ws.readyState === _WebSocket.OPEN) {
+      this.ws.close();
+    }
+
+    this.ws = null;
```

Comment:
Should we set `this.started = false;`?

Reply:
Yes we should, done in 37b7bd4f55083042ec5e4dcbf54b75ccdbd126a9

Reply:
I was reading the code again, and I feel that we call this method from multiple sources. If I got it right, the `this.started` flag indicates whether it should retry to connect if the connection is closed for any reason. I'm reconsidering whether we should really set `this.start = false` here or not. What do you think?

Reply:
Actually, what the reconnection mechanism is doing is calling `setup` again on the `retryConnectionInterval`, setInterval is called [here](https://github.com/HathorNetwork/hathor-wallet-lib/blob/851eec942c1406e9a81c992f5b84f7403fb91dc6/src/websocket/base.ts#L225)

The first thing the `setup` method does is checking that it is not already started, so I think it makes sense to set `this.started` to false

Reply:
You're right! Thanks for the explanation.